httpx/_utils.py

```diff
@@ -240,3 +245,19 @@ def is_ipv6_hostname(hostname: str) -> bool:
     except Exception:
         return False
     return True
+
+
+@asynccontextmanager
+async def safe_async_iterate(
```

Comment:
Interesting. I'm curious about this vs. `aclosing`.

https://docs.python.org/3/library/contextlib.html#contextlib.aclosing

Reply:
I originally went with `aclosing()` but it doesn't work very well with arbitrary async iterables.